### PR TITLE
Include unity activity file upload component (M2-7687)

### PR DESCRIPTION
**Pull request by ramirlm**
- [x] Tests for the changes have been added
- [ ] Related documentation has been added / updated
- [ ] OSS packages added to MindLogger [open source credit page](https://mindlogger.atlassian.net/wiki/spaces/MINDLOGGER1/pages/263127186/Admin+Panel+Applet+Builder+Library)
- [x] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

Adding a file upload component to include a unity file. Depends on [This PR](https://github.com/ChildMindInstitute/mindlogger-backend-refactor/pull/1563)

🔗 [Jira Ticket M2-7687](https://mindlogger.atlassian.net/browse/M2-7687)

### 📸 Screenshots

| Before (Optional)                      | After                                 |
| -------------------------------------- | ------------------------------------- |
|  |  |

### CODE REVIEW

**Review by AlejandroCoronadoN — DISMISSED** [review]

File: src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx [hunk 1]
```diff
@@ -17,8 +17,11 @@ export const Unity = () => {
 
   const [file, setFile] = useState<File | null>(null);
 
-  const urlName = 'activities[0].items[0].config.file';
+  const activities = watch('activities');
+  const unityActivityIndex = activities.findIndex((a: { type: string }) => a.type === 'unity');
```
Comment (ChaconC) [inline src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx]: This can fail if there's more than 1 unity activity

Comment (ramirlm) [reply src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx]: will we have more than one?

Comment (ChaconC) [reply src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx]: It's not decided yet, but multiple activities could be added, each with a different config file.

You can use the `useCurrentActivity` hook to retrieve the correct activity.

```
const { fieldName } = useCurrentActivity();
...
const urlName = `${fieldName}.items.0.config.file`;
```

Comment (ramirlm) [reply src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx]: Thank you!!

File: src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  };
+
+  useEffect(() => {
+    if (!url) return;
+
+    if (url) {
+      trigger(urlName);
+      setFileContent(url);
+    }
+  }, [url]);
```
Comment (ChaconC) [inline src/modules/Builder/features/PerformanceTasks/Unity/Unity.tsx:69]: Please fix `exhaustive-deps` eslint warning.

File: src/modules/Builder/features/PerformanceTasks/Unity/UnityFilePreview/UnityFilePreview.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+interface UnityFilePreviewProps {
+  fileContent: string;
+}
+
+export const UnityFilePreview: React.FC<UnityFilePreviewProps> = ({ fileContent }) => (
```
Comment (ChaconC) [inline src/modules/Builder/features/PerformanceTasks/Unity/UnityFilePreview/UnityFilePreview.tsx:9]: You mentioned creating a new ticket to complete the UI of `UnityFilePreview`, please add this as a TODO in this file

**Review by ChaconC — DISMISSED** [review]
Left a couple of minor comments you should address before merging, but looks good to approve.
Good job @ramirlm!

Edit:
Actually, there's a couple of failing tests so please review those before approving

**Review by ChaconC — APPROVED** [review]

**Pull request merged**